test/package/test_torchscript.py

```diff
@@ -0,0 +1,584 @@
+                return input
+
+        class ModA(torch.nn.Module):
+            def __init__(self, name: str):
+                super().__init__()
+                self.name = name
+
+            def forward(self, input: str):
+                input = input + "_modA:" + self.name
+                return input
+
+        scripted_mod_a = torch.jit.script(ModA("a"))
+        scripted_mod_b = torch.jit.script(ModB("b"))
+        script_mods_list = [scripted_mod_a, scripted_mod_b]
+
+        buffer = BytesIO()
+        with PackageExporter(buffer, verbose=False) as e:
+            e.save_pickle("res", "list.pkl", script_mods_list)
+
+        buffer.seek(0)
+        importer = PackageImporter(buffer)
+        loaded_mod_list = importer.load_pickle("res", "list.pkl")
+        self.assertEqual(loaded_mod_list[0]("input"), scripted_mod_a("input"))
+        self.assertEqual(loaded_mod_list[1]("input"), scripted_mod_b("input"))
+
+    def test_save_shared_scriptmodules(self):
+        """
+        Test saving of single ScriptModule shared by multiple
+        eager modules (ScriptModule should be saved just once).
+        """
+
+        from package_a.test_module import SimpleTest, Mod
+
+        scripted_mod = torch.jit.script(SimpleTest())
+
+        mod1 = Mod(scripted_mod)
+        mod2 = Mod(scripted_mod)
+        mod3 = Mod(scripted_mod)
+
+        buffer = BytesIO()
+        with PackageExporter(buffer, verbose=False) as e:
+            e.save_pickle("res", "mod1.pkl", mod1)
+            e.save_pickle("res", "mod2.pkl", mod2)
+            e.save_pickle("res", "mod3.pkl", mod3)
+
+        buffer.seek(0)
+        importer = PackageImporter(buffer)
+        file_structure = importer.file_structure()
+        self.assertTrue(file_structure.has_file(".data/ts_code/0"))
+        self.assertFalse(file_structure.has_file(".data/ts_code/1"))
+
+    def test_save_shared_tensors(self):
+        """
+        Test that tensors are shared across eager and ScriptModules.
+        """
+        from package_a.test_module import SharedTensorsMod
+
+        class ModA(torch.nn.Module):
+            def __init__(self, x):
+                super().__init__()
+                self.tensor = x
+
+            def forward(self, input):
+                input = input + self.tensor
+                return input
+
+        shared_tensor = torch.rand(2, 3, 4)
+        scripted_mod = torch.jit.script(ModA(shared_tensor))
+
+        mod1 = SharedTensorsMod(shared_tensor, scripted_mod)
+        mod2 = SharedTensorsMod(shared_tensor, scripted_mod)
+
+        buffer = BytesIO()
+        with PackageExporter(buffer, verbose=False) as e:
+            e.save_pickle("res", "mod1.pkl", mod1)
+            e.save_pickle("res", "mod2.pkl", mod2)
+
+        buffer.seek(0)
+        importer = PackageImporter(buffer)
+        loaded_mod_1 = importer.load_pickle("res", "mod1.pkl")
+
+        # assert that there is only one storage stored in package
```

Comment:
same feedback as above. We want to ensure that we only loaded the relevant storages once.